api/pom.xml

```diff
@@ -171,6 +171,16 @@
 				<filtering>true</filtering>
 			</testResource>
 		</testResources>
-	</build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
```

Comment:
Was this intentional?

Reply:
Yeah, add support for java 8 (Lambda and stream API)

Reply:
Is this module meant to run on platform 1.x?

Reply:
From version 1.20.0 of this module will require platform 2.x 
since even for condition resource features, platform 1.x is not implemented and going forward we will support  2.0 and above.

Reply:
Is that something which was discussed and agreed upon? If yes, it would be great pointing me to the meeting notes.

Reply:
I couldn't find the meeting notes but I am sure we had agreed on this to support 2.0 and above or maybe my interpretation is wrong.
Anyway I have refactored java 8 code to 7

Reply:
I heard that too. By my understanding was that, for any new features that we implement, we support platform 2.0 and above, but still allow the module to run on lower platform versions, for the existing old features. For instance, the condition resources that you implemented require higher platform versions, but do not prevent the module from loading on lower platform versions, who may want to use existing features other than conditions. Does this make sense?

Reply:
It does 👍